### DISCUSSION TITLE
How about re-adding Microsoft.Solver.Foundation or better alternative?

### DISCUSSION THREAD

**Discussion by HimanshuBansal112** [dotnet/runtime#84896] [Ideas]
Solving equations with constraints and other problems of equations can be important in many applications. For example, I was making a graph maker for some purpose in c# but I could not find a library for it. Though, there was a library Microsoft.Solver.Foundation which cannot be used in newer versions since outdated. I tried searching several alternative but could not found one that solves equations with constraints in simpler, faster and bug-free method. An in-built method would be a good and reasonable thing. For numbering uses of it, I also was trying to make it for graph such that it will show multiple solutions upto certain limit we want if there are multiple solutions considering every equation and for objective value as well. It would be helpful if there will be a library.

It is moved from https://github.com/dotnet/csharplang/discussions/7125

**Comment by codj0** [comment]
Microsoft.SolverFoundation is an important library .NET Core space needs.
Please port it to .NET 9.

**Reply by Ralf1108** [reply]
Done, see my last reply ;-)

**Comment by huoyaoyuan** [comment]
The library appears to be authored by another research team in Microsoft, not the .NET team.

Trying with the third-party package on NuGet, it requires .NET remoting and fails on .NET Core. Unfortunately there's noting much to do for .NET team since they don't own the source code.

**Reply by HimanshuBansal112** [reply]
I understand. If I have to make all of this from scratch, this will take a lot of time and will not be as good as Microsoft.Solver.Foundation so is there another free library to get same capabilities? And can you please also tell which repository I need to open this request at, so that team can think about adding this?

**Reply by huoyaoyuan** [reply]
> If I have to make all of this from scratch, this will take a lot of time and will not be as good as Microsoft.Solver.Foundation so is there another free library to get same capabilities?

If the license allows, you may do reverse engineering to Microsoft.Solver.Foundation and simply replace the incompatible components.

> And can you please also tell which repository I need to open this request at, so that team can think about adding this?

This is a domain specific library (although math is a very basic domain) and is not in the scope of .NET team. It's open for any person or organization to do their work, including Microsoft Research.
You may go to MathDotNet and see if they have some work related to this area. Remember there's no guarantee for this being in-scope for them.

**Reply by HimanshuBansal112** [reply]
Ok, thanks for your help. I will leave this open if anyone from .Net Team in future, is interested to implement this.

**Comment by Ralf1108** [comment]
Just in case you are still looking for a .NET Standard 2.0 / .NET Core compatible port of Microsoft Solver Foundation library I made one from the last known one and uploaded it to [nuget][1]. Feel free to try it out and report occuring issues.


  [1]: https://www.nuget.org/packages/Reloaded.SolverFoundation/
